impeller/aiks/canvas.cc

```diff
@@ -885,6 +889,28 @@ void Canvas::DrawVertices(const std::shared_ptr<VerticesGeometry>& vertices,
     return;
   }
 
+  // If there is are per-vertex colors, an image, and the blend mode
+  // is simple we can draw without a sub-renderpass.
+  if (blend_mode <= BlendMode::kModulate && vertices->HasVertexColors() &&
```

Comment:
We can do this for the rest of the blend modes once I make some updates to the advanced blends. After that, the only time we'd use the old path is for a gradient/shader color source. There is probably an easier refactor there to eagerly convert those to a texture and then let them go through this code path instead.